build/build.go

```diff
@@ -496,17 +496,14 @@ func (s *Session) BuildPackage(pkg *PackageData) error {
 		return err
 	}
 
-	var jsDecls []*compiler.Decl
 	for _, jsFile := range pkg.JSFiles {
 		code, err := ioutil.ReadFile(filepath.Join(pkg.Dir, jsFile))
 		if err != nil {
 			return err
 		}
-		jsDecls = append(jsDecls, &compiler.Decl{
-			DeclCode: append(append([]byte("\t(function() {\n"), code...), []byte("\n\t}).call($global);\n")...),
-		})
+		code = append(append([]byte("\t(function() {\n"), code...), []byte("\n\t}).call($global);\n")...)
+		pkg.Archive.IncJSCode = append(pkg.Archive.IncJSCode, code...)
```

Comment:
Given than these may be relatively large byte slices, it may help to do this in a more performant way.

I think it'd be more efficient to keep appending to a single byte slice, rather than creating a temporary one.

E.g.:

```Go
pkg.Archive.IncJSCode = append(pkg.Archive.IncJSCode, []byte("\t(function() {\n")...)
pkg.Archive.IncJSCode = append(pkg.Archive.IncJSCode, code...)
pkg.Archive.IncJSCode = append(pkg.Archive.IncJSCode, []byte("\n\t}).call($global);\n")...)
```

I have not benchmarked it, and don't have numbers of how much it's faster or slower, but my intuition tells me is that it should be quite better.